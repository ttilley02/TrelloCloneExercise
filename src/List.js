import React from 'react';
import Card from './Card'
import './List.css';

export default function List(props) {




  return (
    <section className='List'>
      <header className='List-header'>
        <h2>{props.header}</h2>
      </header>
      <div className='List-cards'>
        {props.cards.map((card) =>
          <Card
            id={card.id}
            key={card.id}
            title={card.title}
            content={card.content}
            onClickDelete = {props.onClickDelete}
            
            
          />
        )}
        <button 
          onClick = {() => props.onClickRandom(props.id)}
          type='button'

          className='List-add-button'
        >
          + Add Random Card
        </button>
      </div>
    </section>
  )
}
