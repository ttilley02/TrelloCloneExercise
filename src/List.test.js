import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import List from './List.js';

describe('<List />', () => {
    it('renders without crashing2', () => {
        const cards = [{ id: 'a', title: 'First card', content: 'lorem ipsum' }, { id: 'b', title: '2nd card', content: 'lorem ipsum' }, { id: 'c', title: '3rd card', content: 'lorem ipsum' }];

        // Create a DOM element to render the component into
        const div = document.createElement('div');

        // Render the component
        // If something is wrong it will fail here
        ReactDOM.render(<List cards= {cards} header={"test123"} />, div);

        // Clean up
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders without crashing', () => {
        const cards = [{ id: 'a', title: 'First card', content: 'lorem ipsum' }, { id: 'b', title: '2nd card', content: 'lorem ipsum' }, { id: 'c', title: '3rd card', content: 'lorem ipsum' }];

        // Render the component, as JSON
        const tree = renderer.create(<List cards= {cards} header={"test123"} />).toJSON();
        // Check whether it matches the previous snapshot
        // Stored in __snapshots__/App.test.js.snap
        expect(tree).toMatchSnapshot(); 
    });
});