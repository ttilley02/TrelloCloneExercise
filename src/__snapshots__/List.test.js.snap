// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders without crashing 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      test123
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        First card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        2nd card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        3rd card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
